website/docs/components/form/file-input/partials/code/how-to-use.md

```diff
@@ -0,0 +1,133 @@
+## How to use this component
+
+The File Input component uses the native HTML input with type="file" under the hood but adds visual styling to match other HDS components.
+
+There are two ways to use the File Input component:
+
+- `Form::FileInput::Base`—the Base component: just the `<input type="file">` control.
+- `Form::FileInput::Field`—the Field component: the `<input type="file">` control, with label, helper text, and error messaging (in a wrapping container). 
+
+We recommend using the Field component as it provides built-in accessibility functionality. Use the Base component if you need to achieve custom layouts or for special use cases not covered by the Field component.
+
+{{! ================= }} {{! ===== FIELD ===== }} {{! ================= }}
+
+### Form::FileInput::Field
+
+```handlebars
+<Hds::Form::FileInput::Field as |F|>
+  <F.Label>Upload a file</F.Label>
+</Hds::Form::FileInput::Field>
+```
+
+#### Helper text
+
+You can add extra information to the field using helper text. When helper text is added, the component automatically adds an `aria-describedby` attribute to the input control, associating it with the automatically generated `ID` of the helper text element.
+
+```handlebars
+<Hds::Form::FileInput::Field as |F|>
+  <F.Label>Upload a file</F.Label>
+  <F.HelperText>File size should be a maximum of 2 MB.</F.HelperText>
+</Hds::Form::FileInput::Field>
+```
+
+#### Extra content in label and helper text
+
+!!! Warning
+
+If a link is used within a label, helper text, or error text, it will not be presented as a link to the user with a screen reader; only the text content is read out. As such, care should be taken when considering this feature. If you do need to include a link, include a screen reader-only message that informs the user that some help text includes links, and additional keyboard exploration may be required.
+
+!!!
+
+The `Label` and `HelperText` contextual components used in the Field component yield their content. This means you can also pass structured content.
+
+```handlebars
+<Hds::Form::FileInput::Field as |F|>
+  <F.Label>Upload a file <Hds::Badge @size="small" @text="Beta" /></F.Label>
+  <F.HelperText>This is an experimental feature (<Hds::Link::Inline @href="#">read more</Hds::Link::Inline>).</F.HelperText>
+</Hds::Form::FileInput::Field>
```

Comment:
I like how all examples have relevant content, but I think this one is not something we'd like to recommend as the link text ('read more') is not conformant. We can potentially move the link to more relevant content, or rephrase it somehow.
```suggestion
<Hds::Form::FileInput::Field as |F|>
  <F.Label>Upload a file <Hds::Badge @size="small" @text="Beta" /></F.Label>
  <F.HelperText>This is an <Hds::Link::Inline @href="#">experimental feature</Hds::Link::Inline>.</F.HelperText>
</Hds::Form::FileInput::Field>
```

Reply:
Ok thanks